src/mscorlib/src/System/Array.cs

```diff
@@ -239,12 +212,11 @@ public static void Copy(Array sourceArray, Array destinationArray, int length)
             if (destinationArray == null)
                 ThrowHelper.ThrowArgumentNullException(ExceptionArgument.destinationArray);
             /*
-            Contract.Requires(sourceArray.Rank == destinationArray.Rank);
-            Contract.Requires(length >= 0);
-            Contract.Requires(length <= sourceArray.GetLowerBound(0) + sourceArray.Length);
-            Contract.Requires(length <= destinationArray.GetLowerBound(0) + destinationArray.Length);
+            Debug.Assert(sourceArray.Rank == destinationArray.Rank);
+            Debug.Assert(length >= 0);
+            Debug.Assert(length <= sourceArray.GetLowerBound(0) + sourceArray.Length);
+            Debug.Assert(length <= destinationArray.GetLowerBound(0) + destinationArray.Length);
              */
-            Contract.EndContractBlock();
```

Comment:
Seems like we should either uncomment or delete the whole commented-out section.

Reply:
I'm going to remove them, because this is a public method. Any parameter validation should not be debug only, and hopefully we're doing it already.